### DISCUSSION TITLE
React Hook Form don't register fields properly on Material UI stepper

### DISCUSSION THREAD

**Discussion by ghost** [react-hook-form/react-hook-form#2328] [Q&A] [answered]
I'm using the default example from Material UI steppers, which can be found [here](https://material-ui.com/components/steppers/#non-linear-alternative-label), along with React Hook Form, but when I register the fields, e.g on the first step, they are not submitted when the form is submitted at the last step, but the fields from the last step, where the submit button is, are submitted.

I tried to use the [provider](https://react-hook-form.com/advanced-usage#FormProviderPerformance) but it didn't work, I tried to elevate where I use `useForm()` and use `useCallback(()=> {}, [])` so the reference is not lost between renders, but nothing worked... I have three levels (base component -> modal (tried `useForm()` here / provide the context) -> stepper (tried `useForm()` here - each step is another component) and I provided the context at the right place so I don't understand why it is not storing even though I keep the reference and the data in a context. Any tips?

React Hook Form 6.0.8 and Material UI 5 alpha 4. I used the 6.0.2 and 5 alpha 2 and the same result.

Here's my question on [Stack Over Flow](https://stackoverflow.com/questions/63040096/react-hook-form-dont-register-inputs-when-using-material-ui-stepper).

**Comment by bluebill1049** [comment, marked as answer]
i think you are after `shouldUnregister` config at `useForm()`

**Reply by ghost** [reply]
That worked. Is it possible to validate each field after a certain operation? I saw at the documentation / searched the API and I just found the option `reValidateMode`, which is not exactly what I described.

 Thanks.

**Reply by bluebill1049** [reply]
Take a look at this example: https://codesandbox.io/s/tabs-760h9

**Reply by ghost** [reply]
So its not possible to do that manually, for specific fields using React Hook Form. I'll try to do it using Yup validate function, if this doesn't work I'll try to build one form for each stepper. I appreciate your support, I really do. Thanks

> I saw your video tutorial where you create this stepper, if you see the [STF](https://stackoverflow.com/questions/63040096/react-hook-form-dont-register-inputs-when-using-material-ui-stepper) question you can notice that I put it there as a possible solution to the primary problem described here / there
